ocaml-lsp-server/src/hover_req.ml

```diff
@@ -12,29 +12,40 @@ let environment_mode =
   | Some true -> Extended_variable
   | Some false | None -> Default
 
-let format_contents ~syntax ~markdown ~typ ~doc =
+let print_dividers sections = String.concat ~sep:"\n\n --- \n\n" sections
+
+let format_as_code_block ?(highlighter="ocaml") strings =
```

Comment:
Since this API is private, there's no need to maintain backwards compatibility. Thus, optional arguments aren't particularly useful and just allow the callers not to think about the correct value for this parameter. Making it required will be better